src/DIRAC/Resources/Cloud/CloudEndpoint.py

```diff
@@ -0,0 +1,558 @@
+"""
+   CloudEndpoint is a base class for the clients used to connect to different
+   cloud providers
+"""
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+__RCSID__ = '$Id$'
+
+import os
+import ssl
+import time
+
+from libcloud import security
+from libcloud.compute.types import Provider
+from libcloud.compute.providers import get_driver
+
+# DIRAC
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Utilities.File import makeGuid
+
+from DIRAC.Resources.Cloud.Endpoint import Endpoint
+from DIRAC.Resources.Cloud.Utilities import STATE_MAP
+
+DEBUG = False
+
+
+class CloudEndpoint(Endpoint):
+  """ CloudEndpoint base class
+  """
+
+  def __init__(self, parameters=None):
+    super(CloudEndpoint, self).__init__(parameters=parameters)
+    # logger
+    self.log = gLogger.getSubLogger('CloudEndpoint')
+    self.valid = False
+    result = self.initialize()
+    if result['OK']:
+      self.log.debug('CloudEndpoint created and validated')
+      self.valid = True
+
+  def initialize(self):
+
+    # Relax security
+    security.SSL_VERSION = ssl.PROTOCOL_SSLv23
+    security.VERIFY_SSL_CERT = False
+
+    # Variables needed to contact the service
+    connDict = {}
+    for var in ['ex_domain_name', 'ex_force_auth_url', 'ex_force_service_region', 'ex_force_auth_version',
+                'ex_tenant_name', 'ex_keyname', 'ex_voms_proxy']:
+      if var in self.parameters:
+        connDict[var] = self.parameters[var]
+
+    username = self.parameters.get('User')
+    password = self.parameters.get('Password')
+
+    # log info:
+    if DEBUG:
```

Comment:
It would be better to check what's the current level of `gLogger`

Reply:
This only does an os.system("export ...") if True, which doesn't actually have any effect anyway, so I've removed it entirely.